Sources/SwiftSyntax/SyntaxParser.swift

```diff
@@ -66,7 +69,9 @@ public enum SyntaxParser {
   /// - Throws: `ParserError`
   public static func parse(
     source: String,
-    parseTransition: IncrementalParseTransition? = nil
+    file: String = "",
+    parseTransition: IncrementalParseTransition? = nil,
+    diagConsumer: DiagnosticConsumer? = nil
```

Comment:
Can you make this take a `DiagnosticEngine` instead? An engine might have several consumers that would want to receive diagnostics. You'd probably want to check if _any_ of the consumers requested line and column, and compute it once if so. IMO that would make the parser API a little nicer -- 'parse and push the diagnostics into this diagnostic engine' rather than 'deliver them to this one consumer, of multiple which I might have'.

Also, please name it 'diagnosticEngine' over 'diagEngine'.

Reply:
hmm, we usually need `DiagnosticEngine` for issuing new diagnostics. In here, the clients just want to hear diagnostics so adding `DiagnosticEngine` to the parameter seems to me to complicate the caller site (setting up engine and adding consumers to it). Also, the usefulness of multiple consumers for these diagnostics isn't so clear to me yet.

Reply:
In practice, tools that use SwiftSyntax and deal with diagnostics will create a DiagnosticEngine as part of their initial start-up. If we make this method take a single consumer, that removes the ability for a client to use multiple consumers (say, a linter that wants to print diagnostics in CI and also send them to a web service).

The way this reads now (which I still think is valid) is 'parse this file, and if there are diagnostics, tell this consumer'. I don't think it's unreasonable or more difficult for clients to change that to 'parse this file, and if there are diagnostics, use this engine to diagnose them'.

Reply:
That's fair. I've updated the function signature to use `DiagnosticEngine` instead of `DiagnosticConsumer`.

Reply:
Thank you!